ddl/placement_policy.go

```diff
@@ -223,6 +229,31 @@ func onAlterPlacementPolicy(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 		return ver, errors.Trace(err)
 	}
 
+	is := d.infoCache.GetLatest()
+	ids := is.(infoschema.InfoSchema).DismissPolicyDependencies(oldPolicy.Name.L)
+	if len(ids) != 0 {
+		// build bundle from new placement policy.
+		bundle, err := placement.NewBundleFromOptions(newPolicyInfo.PlacementSettings)
+		if err != nil {
+			return ver, errors.Trace(err)
+		}
+		err = bundle.Tidy()
+		if err != nil {
+			return ver, errors.Trace(err)
+		}
+		// Do the http request only when the rules is existed.
+		if len(bundle.Rules) > 0 {
+			for _, id := range ids {
+				bundle.Reset(id)
+				err = infosync.PutRuleBundles(context.TODO(), []*placement.Bundle{bundle})
+				if err != nil {
```

Comment:
need some rollback logics? If  one of the pd request fails, the previous will still keeps modified configuration.

Reply:
One http request can still have a certain probability that request timeout but succeed in pd side in fact left meta and pd status inconsistency. Roll back is still required, I think. Or we may expect user to fix the inconsistency manually.

Reply:
That is rare. If you say so, we should add rollback for every `PutRuleBundles`, which is not done. Probably create a new issue, throw it in the slack for discussion?

Reply:
I remember it's a old-left problem which the put bundles don't have the atomic attribute.

Reply:
Yes, it is not easy to happen and we will write more codes to make it. I think we can left it in other prs. I'll open a new issue and trace it and then may be we can discuss it later.